src/cost_functions.jl

```diff
@@ -77,20 +80,41 @@ function (f::L2Loss)(sol::DiffEqBase.DESolution)
       end
     end
   end
+  if colloc_grad != nothing
+    dudt = zeros(size(colloc_grad))
+    tmp = zeros(size(colloc_grad)[1])
+    for i = 1:size(colloc_grad)[2]
+      sol.prob.f.f(tmp, sol.u[i], sol.prob.p, sol.t)
+      for j = 1:length(tmp)
+        dudt[j,i] = tmp[j]
+      end
+    end
+    sumsq += sum(sum((dudt - colloc_grad).^2, dims=2), dims=1)[1]
```

Comment:
`sum(abs2,x - y for (x,y) in zip(dudt,colloc_grad))` should work without the allocations.